### PR TITLE
[helm] Fix helm dashboards

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

This PR adds documentation on how to get the included Dashboards to work for a fresh helm installation. It also modifies the local k3d loki cluster to install the necessary relabelings to the `kube-prometheus-stack`, `kube-state-metrics`, and `node-exporter` helm charts so those metrics will work with the included dashboards.
**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- ~[ ] Tests updated~
- [ ] `CHANGELOG.md` updated
- ~[ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`~

### CODE REVIEW

File: production/helm/loki/templates/read/deployment-read.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ spec:
         {{- toYaml .Values.loki.podSecurityContext | nindent 8 }}
       terminationGracePeriodSeconds: {{ .Values.read.terminationGracePeriodSeconds }}
       containers:
-        - name: read
+        - name: loki
```
Comment (jeschkies) [inline production/helm/loki/templates/read/deployment-read.yaml:57]: Why is this change needed?

File: docs/sources/installation/helm/monitor-and-alert/index.md [hunk 2]
```diff
@@ -27,8 +26,63 @@ Rules and alerts are automatically deployed.
...
+      - action: replace
+        replacement: <CLUSTER_NAME>
+        targetLabel: cluster
+```
+
+The `kube-prometheus-stack` installs `ServicMonitor` and `PrometheusRule` resources for monitoring Kubernetes, and it depends on the `kube-state-metrics` and `prometheus-node-exporter` helm charts which also install `ServiceMonitor` resources for collecting `kubelet` and `node-exporter` metrics. The above values file adds the necessary additional relabelings are required for these metrics to work with the included dashboards.
+
+If using this helm chart in an environment which does not allow for the installation of `kube-prometheus-stack` or custom CRDs, it is recommended you run `helm template` on the `kube-prometheus-stack` helm chart with the above values file, and grab all generated `ServiceMonitor` and `PrometheusRule` resources. Note those resources may have to be modified with the correct ports and selectors to find the various services such as `kubelet` and `node-exporter` in your environment.
 
 **To install the dashboards:**
```
Comment (jeschkies) [inline docs/sources/installation/helm/monitor-and-alert/index.md:87]: I thinks this section needs some details on how the dashboards in the `ConfigMap` must be mounted.

**Review by jeschkies — CHANGES_REQUESTED** [review]

File: docs/sources/installation/helm/monitor-and-alert/index.md [hunk 3]
```diff
@@ -27,8 +26,63 @@ Rules and alerts are automatically deployed.
...
+  monitor:
+    relabelings:
+      - action: replace
+        replacement: <CLUSTER_NAME>
+        targetLabel: cluster
+```
+
+The `kube-prometheus-stack` installs `ServicMonitor` and `PrometheusRule` resources for monitoring Kubernetes, and it depends on the `kube-state-metrics` and `prometheus-node-exporter` helm charts which also install `ServiceMonitor` resources for collecting `kubelet` and `node-exporter` metrics. The above values file adds the necessary additional relabelings are required for these metrics to work with the included dashboards.
+
+If using this helm chart in an environment which does not allow for the installation of `kube-prometheus-stack` or custom CRDs, it is recommended you run `helm template` on the `kube-prometheus-stack` helm chart with the above values file, and grab all generated `ServiceMonitor` and `PrometheusRule` resources. Note those resources may have to be modified with the correct ports and selectors to find the various services such as `kubelet` and `node-exporter` in your environment.
```
[the remark this answers is not in the document]
Comment (trevorwhitney) [reply docs/sources/installation/helm/monitor-and-alert/index.md:85]: The metrics that come from kuberenetes come from services like `kubelet` and processes like `node-exporter` and `kube-state-metrics`. Those things need to be scraped by Prometheus, this is explaining how those scrape configurations get configured.

File: production/helm/loki/templates/read/deployment-read.yaml (hunk 1 above)
Comment (trevorwhitney) [reply production/helm/loki/templates/read/deployment-read.yaml:57]: There's a few queries in our dashboards that are looking for `container="loki"`

File: docs/sources/installation/helm/monitor-and-alert/index.md (hunk 2 above)
Comment (trevorwhitney) [reply docs/sources/installation/helm/monitor-and-alert/index.md:87]: We have that further down:

```markdown
**To install the dashboards:**

1. Dashboards are enabled by default. Set `monitoring.dashboards.namespace` to the namespace of the Grafana instance if it is in a different namespace than this Loki cluster.
```

Are you suggesting we move that section up here?

File: docs/sources/installation/helm/monitor-and-alert/index.md [hunk 4]
```diff
@@ -27,8 +26,63 @@ Rules and alerts are automatically deployed.
 - Helm 3 or above. See [Installing Helm](https://helm.sh/docs/intro/install/).
 - A running Kubernetes cluster with a running Loki deployment.
 - A running Grafana instance.
-- A running Prometheus operator in order for recording rules for the dashboards
-  to work.
+- A running Prometheus Operator installed using the `kube-prometheus-stack` Helm chart.
+
+**Prometheus Operator Prequisites**
+
+The included dashboards require certain metric relabelings be applied to Kubernetes metrics. The best way to accomplish this is to install the `kube-prometheus-stack` Helm chart with the following values file, replacing `CLUSTER_NAME` with the name of your cluster. The cluster name is what you specify during the helm installation, so a cluster installed with the command `helm install loki-cluster grafana/loki` would be called `loki-cluster`.
```
Comment (JStickler) [inline docs/sources/installation/helm/monitor-and-alert/index.md:33]:
```suggestion
The dashboards require certain metric labels to display Kubernetes metrics. The best way to accomplish this is to install the `kube-prometheus-stack` Helm chart with the following values file, replacing `CLUSTER_NAME` with the name of your cluster. The cluster name is what you specify during the helm installation, so a cluster installed with the command `helm install loki-cluster grafana/loki` would be called `loki-cluster`.
```

File: docs/sources/installation/helm/monitor-and-alert/index.md [hunk 5]
```diff
@@ -27,8 +26,63 @@ Rules and alerts are automatically deployed.
...
+
+prometheus:
+  monitor:
+    relabelings:
+      - action: replace
+        replacement: <CLUSTER_NAME>
+        targetLabel: cluster
+```
+
+The `kube-prometheus-stack` installs `ServicMonitor` and `PrometheusRule` resources for monitoring Kubernetes, and it depends on the `kube-state-metrics` and `prometheus-node-exporter` helm charts which also install `ServiceMonitor` resources for collecting `kubelet` and `node-exporter` metrics. The above values file adds the necessary additional relabelings are required for these metrics to work with the included dashboards.
```
Comment (JStickler) [inline docs/sources/installation/helm/monitor-and-alert/index.md:83]:
```suggestion
The `kube-prometheus-stack` installs `ServicMonitor` and `PrometheusRule` resources for monitoring Kubernetes, and it depends on the `kube-state-metrics` and `prometheus-node-exporter` helm charts which also install `ServiceMonitor` resources for collecting `kubelet` and `node-exporter` metrics. The above values file adds the necessary additional labels required for these metrics to work with the included dashboards.
```

File: docs/sources/installation/helm/monitor-and-alert/index.md [hunk 6]
```diff
@@ -27,8 +26,63 @@ Rules and alerts are automatically deployed.
...
+
+If using this helm chart in an environment which does not allow for the installation of `kube-prometheus-stack` or custom CRDs, it is recommended you run `helm template` on the `kube-prometheus-stack` helm chart with the above values file, and grab all generated `ServiceMonitor` and `PrometheusRule` resources. Note those resources may have to be modified with the correct ports and selectors to find the various services such as `kubelet` and `node-exporter` in your environment.
```
Comment (JStickler) [inline docs/sources/installation/helm/monitor-and-alert/index.md:85]:
```suggestion
If you are using this helm chart in an environment which does not allow for the installation of `kube-prometheus-stack` or custom CRDs, you should run `helm template` on the `kube-prometheus-stack` helm chart with the above values file, and grab all generated `ServiceMonitor` and `PrometheusRule` resources. These resources may have to be modified with the correct ports and selectors to find the various services such as `kubelet` and `node-exporter` in your environment.
```
And I want another word for "grab", but I'm not sure what's appropriate here.  "note"? "collect"?

**Review by JStickler — COMMENTED** [review]
[Docs review] Couple of small suggestions

File: docs/sources/installation/helm/monitor-and-alert/index.md (hunk 6 above)
Comment (trevorwhitney) [reply docs/sources/installation/helm/monitor-and-alert/index.md:85]: how about "review"?

Comment (JStickler) [reply docs/sources/installation/helm/monitor-and-alert/index.md]: That works.

File: docs/sources/installation/helm/monitor-and-alert/index.md (hunk 2 above)
Comment (jeschkies) [reply docs/sources/installation/helm/monitor-and-alert/index.md:87]: No, I think it's missing the information how Grafana needs to be configured to pick up the dashboards. Like you did here https://github.com/grafana/loki/blob/main/tools/dev/k3d/environments/helm-cluster/main.jsonnet#L90

**Review by jeschkies — APPROVED** [review]

File: docs/sources/installation/helm/monitor-and-alert/index.md (hunk 2 above)
Comment (trevorwhitney) [reply docs/sources/installation/helm/monitor-and-alert/index.md:87]: Now I understand! Added 2 more steps on how to mount and provision the dashboards in the Grafana container.
